models/progresses.js

```diff
@@ -47,9 +48,14 @@ const createProgressDocument = async (progressData) => {
  * @throws {Error} If the userId or taskId is invalid or does not exist.
  **/
 const getProgressDocument = async (queryParams) => {
+  const { dev } = queryParams;
   await assertUserOrTaskExists(queryParams);
   const query = buildQueryToFetchDocs(queryParams);
   const progressDocs = await getProgressDocs(query);
+
+  if (dev === "true") {
+    return await addUserDetailsToProgressDocs(progressDocs);
+  }
```

Comment:
instead of checking dev here, do an early return before all this. will save you from fetching all this.

Reply:
The reason for this structure is that we want to run all the common logic (like fetching the progress docs) regardless of the dev flag. Only if dev is true, we add the extra step of adding userdata with addUserDetailsToProgressDocs.